layers/+lang/go/packages.el

```diff
@@ -37,12 +38,14 @@
 (defun go/post-init-flycheck ()
   (spacemacs/add-flycheck-hook 'go-mode))
 
-(defun go/init-go-mode()
-  (when (memq window-system '(mac ns x))
-    (dolist (var '("GOPATH" "GO15VENDOREXPERIMENT"))
-      (unless (getenv var)
-        (exec-path-from-shell-copy-env var))))
+(defun go/pre-init-exec-path-from-shell ()
+  (spacemacs|use-package-add-hook exec-path-from-shell
+    :pre-config
+    (dolist (var '("GOPATH" "GO15VENDOREXPERIMENT") exec-path-from-shell-variables)
```

Comment:
For some reason, returning `nil` here stops the package configuration. Is that expected?